src/bin/pallene-debug

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env lua
+
+local argparse = require "argparse"
+
+local opts
+do
+    local p = argparse("pallene-debug", "Pallene debugger script for call-stack backtrace")
+
+    p:argument("lua_script", "Lua file to debug with Pallene module")
+    p:argument("args",       "Arguments passed to Lua script")       :args("*")
+
+    opts = p:parse()
+end
+
+local fn, err = loadfile(opts.lua_script)
+
+-- Was loading and parsing the file successful?
+if not fn then
+    io.stderr:write("pallene-debug: "..err.."\n")
+    os.exit(1)
+end
+
+-- Supress I/O.
+local original_print = print
+local original_io_write = io.write
+
+print    = function() end
+io.write = function() end
+
+-- Run the script initially so that we can figure out whether
+-- we have Pallene modules that is compiled with `--use-traceback` flag.
+local _, _ = pcall(fn)
```

Comment:
Running the script twice sounds sketchy. It can have other side effects other than printing to screen, and it will take twice as long.

I suppose it would make more sense for the pallene-debug script to supply the traceback function, instead of getting it from a global variable in the loaded script.

Maybe what we should do is create a standalone C module that exports the Pallene traceback and install it together with pallenec. Then, in this script we can require the traceback function.

Reply:
Say the Lua script didn't use any Pallene module with traceback. What then? If we supply our own traceback function it might segfault because there is no stack to begin with. That would be risky. We need to make sure somehow that a Pallene module with traceback exists.

Reply:
Could the traceback function check the Lua registry to see if the Pallene stack exists?

Reply:
It could.

Maybe it's a good idea to detach the traceback fn once and for all. Because we are using it explicitly anyway. But when calling with `xpcall`, can the traceback fn return anything? Maybe we can introduce another fn for that?

Reply:
I don't understand the "when calling with xpcall" question.

Reply:
The traceback fn should return anything to denote there is no stack to work on. This way we would be able to warn the users to use the `--use-traceback` flag. It would be better than not printing any traceback when there is no Pallene stack at all.

My question was if we pass our traceback to `xpcall` as second argument can we fetch what it returned? Or we have to introduce another fn for it?

Reply:
On the contrary, I have a grosshack (sorta) prepared. Though we have to run Lua scripts twice, it won't be slow. Also it shouldn't have side effects.

Creating a separate module, exporting traceback fn is way to hassle for a single debug script.